configure.ac

```diff
@@ -224,66 +227,60 @@ case "$host_os" in
 esac
 
 # Checks for header files.
-AC_CHECK_HEADERS([fcntl.h])
-AC_CHECK_HEADERS([stdlib.h])
-AC_CHECK_HEADERS([string.h])
-AC_CHECK_HEADERS([strings.h])
-AC_CHECK_HEADERS([sys/ioctl.h])
-AC_CHECK_HEADERS([syslog.h])
-AC_CHECK_HEADERS([unistd.h])
-AC_CHECK_HEADERS([netinet/in.h])
-AC_CHECK_HEADERS([sys/socket.h])
-AC_CHECK_HEADERS([arpa/inet.h])
-AC_CHECK_HEADERS([netdb.h])
-AC_CHECK_HEADERS([limits.h])
-AC_CHECK_HEADERS([stdint.h])
 AC_CHECK_HEADERS([sys/epoll.h])
+AC_CHECK_FUNCS([kevent])
+# if neither sys/epoll.h nor kevent are present, we should fail.
+
+if test "x$ac_cv_header_sys_epoll_h" = xno && test "x$ac_cv_func_kevent" = xno; then
+	AC_MSG_ERROR([Both epoll and kevent unavailable on this OS])
+fi
+
+if test "x$ac_cv_header_sys_epoll_h" = xyes && test "x$ac_cv_func_kevent" = xyes; then
+	AC_MSG_ERROR([Both epoll and kevent available on this OS, please contact the maintainers to fix the code])
+fi
 
 if test "x$enable_libknet_sctp" = xyes; then
 	AC_CHECK_HEADERS([netinet/sctp.h],, [AC_MSG_ERROR(["missing required SCTP headers"])])
 fi
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_INLINE
-AC_TYPE_SIZE_T
 AC_TYPE_PID_T
+AC_TYPE_SIZE_T
 AC_TYPE_SSIZE_T
 AC_TYPE_UINT8_T
 AC_TYPE_UINT16_T
 AC_TYPE_UINT32_T
 AC_TYPE_UINT64_T
+AC_TYPE_INT8_T
+AC_TYPE_INT16_T
 AC_TYPE_INT32_T
+AC_TYPE_INT64_T
 
-# Checks for library functions.
-AC_FUNC_ALLOCA
-AC_FUNC_FORK
-AC_FUNC_MALLOC
-AC_FUNC_REALLOC
-AC_CHECK_FUNCS([kevent])
-# if neither sys/epoll.h nor kevent are present, we should fail.
-
-if test "x$ac_cv_header_sys_epoll_h" = xno && test "x$ac_cv_func_kevent" = xno; then
-	AC_MSG_ERROR([Both epoll and kevent unavailable on this OS])
-fi
-
-if test "x$ac_cv_header_sys_epoll_h" = xyes && test "x$ac_cv_func_kevent" = xyes; then
-	AC_MSG_ERROR([Both epoll and kevent available on this OS, please contact the maintainers to fix the code])
+if test "x$enable_man" = "xyes"; then
+	AC_ARG_VAR([DOXYGEN_PATH], [override doxygen executable])
+	if test "x$DOXYGEN_PATH" = "x"; then
+		DOXYGEN_PATH=doxygen
+	fi
+	AC_CHECK_PROGS([DOXYGEN], [$DOXYGEN_PATH], [no])
```

Comment:
Hi Fabio, why not simply

    AC_ARG_VAR([DOXYGEN], [override doxygen executable])
    AC_CHECK_PROGS([DOXYGEN], [doxygen], [no])
?

Reply:
First time using AC_ARG_VAR, the few examples I found around did check the result and set a default. Pure and simple ignorance on my side ;)